lib/encoder.js

```diff
@@ -1442,7 +1442,7 @@ function handleBufferRef(buffer) {
  * @private
  */
 function decodeCollectionLengthV3(bytes, offset) {
-  return bytes.readUInt32BE(offset);
+  return bytes.readInt32BE(offset);
```

Comment:
Noticed that this is actually a signed integer ([java driver code](https://github.com/datastax/java-driver/blob/3.x/driver-core/src/main/java/com/datastax/driver/core/CodecUtils.java#L63-L75), [cassandra code](https://github.com/apache/cassandra/blob/cassandra-3.0/src/java/org/apache/cassandra/transport/CBUtil.java#L417)) although not sure if this actually matters since collections shouldn't contain null values.